src/particle_filter/include/particle_filter/particleFilter6DOF.h

```diff
@@ -0,0 +1,51 @@
+#ifndef PARTICLE_FILTER_6DOF_H
+#define PARTICLE_FILTER_6DOF_H
+
+class particleFilter
+{
+ public:
+  static const int cdim = 6;
+  typedef double cspace[cdim]; // configuration space of the particles
+
+  particleFilter (int n_particles, cspace b_init[2], 
+		  double Xstd_ob=0.0001, double Xstd_tran=0.0025,
+		  double Xstd_scatter=0.0001, double R=0.01);
+
+  void addObservation (double obs[3], double cube[3], distanceTransform *dist_transform, int idx_obs);
```

Comment:
Use 
distanceTransform &dist_transform 
to pass by reference without needed to use pointers